docs/index.rst

```diff
@@ -106,29 +106,85 @@ You can use pip and make to quickly set up a development environment::
+* Maintain Coding Standards
+    * Keep pep8 rules
+    * Follow the same stylistic and logical patterns used in the code
+        * All entity class names and class attributes have to be in the singular format
+        * All required entity attributes have to have `required=True` parameter
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to review those failures and fix them in the same PR's or raise another. The tracking of failures would be responsibility of contributor.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
+* To make the review process easy for all reviewers and anyone else interested in the new functionality, please provide some output making use of your changes. Having example of usage in docstring along with your code could really help others to build up on your code. You can add log from Python interactive shell or some tests results (from Robottelo / Foreman Ansible Modules) in PR message, or you can do something completely different - as long as it runs your code, it's fine!
+* If PR is applicable for many branches (e.g. master and one of '6.X.z' branches), specify that information in PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
+* When in doubt, ask on IRC. Join #robottelo on `freenode`_
+
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    if possible, the contributor should set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
+
+* Test results from upstream devel or from upstream nightly
+    The API call results are required to make the review process more firm.The PR author has to provide results from any library that uses the contributed code by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable
```

Comment:
Sorry, but I have to disagree here. I provided detailed explanation in https://github.com/SatelliteQE/nailgun/pull/600#discussion_r257207492 why we can't simply `s/can/has` in second sentence.

Second proposal from Alex is fine for me. If you agree, make line 140:

```
The API call results are required from PR author to make the review process more firm. Author can provide results from any library that uses the contributed code by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable as well.
```

Reply:
Thanks for updating that, @san7ket !